responses/__init__.py

```diff
@@ -1031,9 +1032,17 @@ def _on_request(
                     response=response,  # type: ignore[misc]
                 )
                 return self._on_request(adapter, request, retries=retries, **kwargs)
-            except MaxRetryError:
+            except MaxRetryError as e:
                 if retries.raise_on_status:
-                    raise
+                    """Since we call 'retries.increment()' by ourselves, we always set "error"
+                    argument equal to None, thus, MaxRetryError exception will be raised with
+                    ResponseError as a 'reason'. That is why following 'if' condition is commented
+                    out. As reference see 'requests' implementation via:
+                    https://github.com/psf/requests/blob/
+                    177dd90f18a8f4dc79a7d2049f0a3f4fcc5932a0/requests/adapters.py#L549"""
+                    # if isinstance(e.reason, ResponseError):
+                    raise RetryError(e, request=request)
```

Comment:
```suggestion
                    """Since we call 'retries.increment()' by ourselves, we always set "error"
                    argument equal to None, thus, MaxRetryError exception will be raised with
                    ResponseError as a 'reason'. 
                    
                    Here we're emulating the `if isinstance(e.reason, ResponseError):`
                    branch found at:
                    https://github.com/psf/requests/blob/
                    177dd90f18a8f4dc79a7d2049f0a3f4fcc5932a0/requests/adapters.py#L549"""
                    raise RetryError(e, request=request)
```

If we reword this we don't need to have the commented code and a comment on why we have commented out code.

Reply:
@markstory thank you!